### ISSUE TITLE
[Issue]: Save button still broken

### ISSUE THREAD

**Issue by afbagwell** [vladmandic/automatic#2223]
### Issue Description

Ok, I'm making an issue this time! :)

I'm still getting the save button error. And yet I've also seen it work at least once. I'm not sure what's going on. It is also affecting the save button in the image browser following a run.

Also, the box below the buttons usually prints out the file name that has been saved. Instead it's frozen on a file icon.

<img width="946" alt="Screenshot 2023-09-18 195126" src="https://github.com/vladmandic/automatic/assets/1495201/a333d636-9be7-4b0b-99a3-6ae4ad731a7b">

### Version Platform Description

Starting SD.Next
Python 3.10.12 on Linux
Version: app=sd.next updated=2023-09-18 hash=8797a34e
              url=git@github.com:vladmandic/automatic.git/tree/master
Platform: arch=x86_64 cpu=x86_64 system=Linux release=5.15.90.1-microsoft-standard-WSL2 python=3.10.12
nVidia CUDA toolkit detected
Extensions enabled: ['Lora', 'sd-webui-agent-scheduler', 'stable-diffusion-webui-rembg',
                         'SwinIR', 'multidiffusion-upscaler-for-automatic1111', 'stable-diffusion-webui-images-browser',
                         'clip-interrogator-ext', 'a1111-sd-webui-lycoris', 'LDSR', 'ScuNET', 'sd-webui-controlnet',
                         'sd-extension-system-info', 'sd-dynamic-thresholding', 'sd-canvas-editor',
                         'sd-webui-inpaint-anything', 'ultimate-upscale-for-automatic1111', 'canvas-zoom',
                         'sd-webui-segment-anything', 'adetailer', 'sd-webui-reactor', 'sd-webui-regional-prompter',
                         'sd-dynamic-prompts', 'openpose-editor', 'sd-webui-aspect-ratio-helper', 'sd-face-editor']
Engine: backend=Backend.ORIGINAL compute=cuda mode=no_grad device=cuda
Device: device=NVIDIA GeForce RTX 4090 n=1 arch=sm_90 cap=(8, 9) cuda=11.8 cudnn=8904 driver=531.79

Firefox 117.0.1 (64-bit)

### Relevant log output

```shell
│ /home/afbag/sdnext/modules/call_queue.py:58 in f                                                                     │
│                                                                                                                      │
│   57 │   │   │   │   pr.enable()                                                                                     │
│ ❱ 58 │   │   │   res = func(*args, **kwargs)                                                                         │
│   59 │   │   │   if res is None:                                                                                     │
│                                                                                                                      │
│ /home/afbag/sdnext/modules/ui_common.py:113 in save_files                                                            │
│                                                                                                                      │
│   112 │   │   │   if shared.opts.use_save_to_dirs_for_ui:                                                            │
│ ❱ 113 │   │   │   │   namegen = modules.images.FilenameGenerator(p, seed=p.all_seeds[i], promp                       │
│   114 │   │   │   │   dirname = namegen.apply(shared.opts.directories_filename_pattern or "[pr                       │
│                                                                                                                      │
│ /home/afbag/sdnext/modules/images.py:328 in __init__                                                                 │
│                                                                                                                      │
│   327 │   │   self.image = image                                                                                     │
│ ❱ 328 │   │   self.index = index if self.p is None or self.p.batch_size == 1 else self.p.batch                       │
│   329                                                                                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'PObject' object has no attribute 'batch_index'
```

### Backend

Original

### Model

SD 1.5

### Acknowledgements

- [X] I have read the above and searched for existing issues
- [X] I confirm that this is classified correctly and its not an extension issue

**Comment by vladmandic** [comment]
second attempt of fixing pushed :)

**Comment by afbagwell** [comment]
Found another save button problem. This time its the save button inside the viewer instance provided by the Agent Scheduler. I ran several batches overnight, ran through them this morning with the viewer, hit the save button for the ones I wanted to keep, then deleted all the generations from their initial folder.

Just a few minutes ago I verified that none of the images saved this way actual get copied. No "Copying image: ..." output in the console. I lost about 60 images this way.

If you use the image viewer from the "From Text" tab it works fine though.
